core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala

```diff
@@ -322,18 +326,19 @@ class ReassignPartitionsCommand(zkUtils: ZkUtils, proposedAssignment: Map[TopicA
     * Limit the throttle on currently moving replicas. Note that this command can use used to alter the throttle, but
     * it may not alter all limits originally set, if some of the brokers have completed their rebalance.
     */
-  def maybeLimit(throttle: Long) {
-    if (throttle >= 0) {
+  def maybeLimit(throttle: Throttle) {
+    if (throttle.value >= 0) {
       val existingBrokers = existingAssignment().values.flatten.toSeq
       val proposedBrokers = proposedAssignment.values.flatten.toSeq
       val brokers = (existingBrokers ++ proposedBrokers).distinct
 
       for (id <- brokers) {
         val configs = AdminUtils.fetchEntityConfig(zkUtils, ConfigType.Broker, id.toString)
-        configs.put(DynamicConfig.Broker.LeaderReplicationThrottledRateProp, throttle.toString)
-        configs.put(DynamicConfig.Broker.FollowerReplicationThrottledRateProp, throttle.toString)
+        configs.put(DynamicConfig.Broker.LeaderReplicationThrottledRateProp, throttle.value.toString)
+        configs.put(DynamicConfig.Broker.FollowerReplicationThrottledRateProp, throttle.value.toString)
         AdminUtils.changeBrokerConfig(zkUtils, Seq(id), configs)
       }
+      Thread.sleep(throttle.postUpdateDelay)
```

Comment:
If the rebalance kicks in before the throttle is set data isn't replicated. The current solution is to just make the test run for longer, with more data, so it averages out. 

It's in response to this commit: https://github.com/apache/kafka/commit/5e4600ae469559b9cdb36c17efb90cc6b6388826